src/Conversion/ONNXToTorch/NN/GlobalAveragePool.cpp

```diff
@@ -80,49 +54,22 @@ struct ONNXGlobalAveragePoolOpToTorchLowering : public ConversionPattern {
   LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
       ConversionPatternRewriter &rewriter) const final {
 
-    ONNXGlobalAveragePoolOp op1 = 
-	    llvm::dyn_cast<ONNXGlobalAveragePoolOp>(op);
-    ONNXGlobalAveragePoolOpAdaptor adapter(op1);
-    mlir::MLIRContext *context = op1.getContext();
-    Location loc = op1.getLoc();
-
-    Value x = op1.X(); // ONNX operands
-
-    auto ty = IntegerType::get(op1.getContext(), 64);
-    auto inputShape = x.getType().cast<ShapedType>().getShape();
-    int64_t inputRank = inputShape.size();
-    auto f1 = IntegerAttr::get(ty, inputRank);
-    Value f1v = rewriter.create<ConstantIntOp>(loc, f1);
-
-    TensorType x_tensor_type = x.getType().cast<TensorType>();
-    TensorType op_tensor_type = 
-	    op->getResult(0).getType().cast<TensorType>();
-
-    auto xTy = Torch::ValueTensorType::get(
-        context, x_tensor_type.getShape(), x_tensor_type.getElementType());
-    auto resultTy = Torch::ValueTensorType::get(op1.getContext(),
-        op_tensor_type.getShape(), op_tensor_type.getElementType());
-    auto xtt = rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(
-        loc, xTy, x);
+    ONNXGlobalAveragePoolOp globalAveragePool =
+        llvm::dyn_cast_or_null<ONNXGlobalAveragePoolOp>(op);
 
-    llvm::outs() << "\n resultTy:"
-                 << "\n"
-                 << resultTy << "\n"
-                 << "\n";
-    llvm::outs() << "xtt torch tensor from MLIR tensor:"
-                 << "\n"
-                 << xtt << "\n"
-                 << "\n";
+    assert(globalAveragePool && "Expecting op to have a strong type");
 
-    Value atenGlobAvgpool2d =
-        rewriter.create<AtenAdaptiveAvgPool2dOp>(loc, resultTy, xtt, f1v);
+    mlir::MLIRContext *context = globalAveragePool.getContext();
+    Location loc = globalAveragePool.getLoc();
 
-    llvm::outs() << "AtenAdaptiveAvgPool2dOp operation creation"
-                 << "\n"
-                 << atenGlobAvgpool2d << "\n"
-                 << "\n";
+    auto x = globalAveragePool.X();
+    auto resultType =
+        toTorchType(context, globalAveragePool.getResult().getType());
+    auto xTensor = getTorchTensor(x, rewriter, context, loc);
+    auto rank = getRank(x, rewriter, context, loc);
 
-    Value result = atenGlobAvgpool2d;
+    Value result = rewriter.create<AtenAdaptiveAvgPool2dOp>(
+        loc, resultType, xTensor, rank);
```

Comment:
I created a sample Colab notebook that illustrates this issues: https://colab.research.google.com/drive/1P7tdabLQh87vwoVEzU9-Qivz21V9Ryn4?usp=sharing

I wrote a PyTorch module that, when converted to onnx generates a `GlobalAvgPool` op, but when converted to torch-mlir, generates a `aten.AdaptiveAvgPool2d` op with an output size of 1 (independent of the rank of the input tensor.

Reply:
cc @anushree1998

Reply:
(sorry I had accidentally changed the colab that I sent in the previous link, I've fixed it now. Just in case someone looked at the link and was confused why the torch-mlir wasn't outputting an AdaptiveAvgPool).

Reply:
@siboehm, we tried running it from this branch, `onnx-mlir --EmitONNXIR --debug  onnx_model.onnx`, we are still getting `onnx.ReduceMean`. 

If I run it with `onnx-mlir --EmitONNXIR --debug  onnx_model.onnx --run-torch-pass`, I get `torch.aten.mean`.

Instead of the rank, now we are passing as list: `[1, 1]`.

Reply:
Maybe this has changed w/ different versions of Pytorch / ONNX. If the output shape is correct now, then I'm happy with this PR once a test has been added.

Reply:
Hi @siboehm,
This is the result of our model. It also emitting 1:
input: "input"
output: "1"
name: "GlobalAveragePool_0"
op_type: "GlobalAveragePool"